### PR TITLE
ncm-mysql: Schema fixes for `component_mysql_db_user` type

### PR DESCRIPTION
**Pull request by jrha**
See #250 for background.

### CODE REVIEW

**Comment by stdweird** [comment]
retest this please

File: ncm-mysql/src/main/pan/components/mysql/schema.pan [hunk 1]
```diff
@@ -35,8 +35,8 @@ function component_mysql_check_db_script = {
 };
 
 type component_mysql_db_user = {
-  'password'  : string
-  'rights'    : string[]
+  'password'  : string with match(SELF, '^[^\\]+$')
+  'rights'    : string[] = list('ALL PRIVILEGES')
```
Comment (stdweird) [inline ncm-mysql/src/main/pan/components/mysql/schema.pan]: is `ALL PRIVILIGES` really the default in mysql? it seems very very wrong (and maybe we should settle for something lower; like readonly?)

Comment (stdweird) [inline ncm-mysql/src/main/pan/components/mysql/schema.pan]: also something restricted like
```pan
type component_mysql_user_right = string with match(SELF, '^(ALL( PRIVILEGES)?|ALTER( ROUTINE)?|CREATE( (ROUTINE|TEMPORARY TABLES|USER|VIEW))?|DELETE|DROP|EVENT|EXECUTE|FILE|GRANT OPTION|INDEX|INSERT|LOCK TABLES|PROCESS|REFERENCES|RELOAD|REPLICATION (CLIENT|SLAVE)|SELECT|SHOW (DATABASES|VIEW)|SHUTDOWN|SUPER|TRIGGER|UPDATE|USAGE)$');

    'rights' : component_mysql_user_right[] = list('SELECT')
...
```
(completely untested)

Comment (jrha) [inline ncm-mysql/src/main/pan/components/mysql/schema.pan]: I don't know, it was your suggestion...

Comment (jrha) [inline ncm-mysql/src/main/pan/components/mysql/schema.pan]: `SELECT` might be a better default?

**Comment by jrha** [comment]
I checked your code, it seemed fine so I've added it :grin:

**Comment by stdweird** [comment]
@jrha thanks for the extended testing :grinning:

**Comment by jrha** [comment]
@jouvin, could you take a look at this and merge if happy?

**Pull request merged**